platform/ios/MGLMapboxEvents.m

```diff
@@ -655,6 +657,41 @@ - (NSString *) deviceOrientation {
 
 // Can be called from any thread.
 //
+- (NSString *) applicationState {
+    __block NSString *result;
+
+    NSString *(^applicationStateBlock)(void) = ^{
+        switch ([UIApplication sharedApplication].applicationState) {
+            case UIApplicationStateActive:
+                result = @"Active";
+                break;
+            case UIApplicationStateInactive:
+                result = @"Inactive";
+                break;
+            case UIApplicationStateBackground:
+                result = @"Background";
+                break;
+            default:
+                result = @"Default - Unknown";
```

Comment:
The string keys don’t need to be user-presentable; the backend would take care of that.